### PR TITLE
test: use random ports in integration tests

### PR DESCRIPTION
**Pull request by fiam**
Use wgc to generate the configuration with the random ports on demand

## Motivation and Context

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

**Review by jensneuse — APPROVED** [review]

File: router-tests/go.mod [hunk 1]
```diff
@@ -4,13 +4,16 @@ replace github.com/wundergraph/cosmo/router => ../router
 
 replace github.com/wundergraph/cosmo/demo => ../demo
 
-go 1.20
+go 1.21
+
+toolchain go1.21.2
```
Comment (StarpTech) [inline router-tests/go.mod:9]: This produces a build error. What does bring us?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: router-tests/runner/runner.go [hunk 2]
```diff
@@ -88,3 +104,22 @@ func Wait(ctx context.Context, r SubgraphsRunner) error {
 	}
 	return nil
 }
+
+func randomFreePorts() *subgraphs.Ports {
+	ports := make([]int, 4)
+	for ii := range ports {
```
Comment (StarpTech) [inline router-tests/runner/runner.go:110]: Just `i`?

**Review by StarpTech — CHANGES_REQUESTED** [review]

**Review by StarpTech — APPROVED** [review]
LGTM

**Comment by fiam** [comment]
This one needs #187 and a release because we're using absolute paths for the wgc output file

**Pull request merged**